### DISCUSSION TITLE
question/promise-executor-ii

### DISCUSSION THREAD

**Discussion by giscus** [us-fe/us-fe.github.io#51] [Announcements]
# question/promise-executor-ii

北美前端面试攻略

https://us-fe.github.io/question/promise-executor-ii.html

**Comment by xiaoyu2er** [comment]
答案 

```js
false
true
false
false
```

这里涉及的知识点

+ Promise.resolve 如果 参数是一个 promise 本身，那么则立即返回 https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve#%E8%BF%94%E5%9B%9E%E5%80%BC
+ new Promise 返回一个新的 Promise
+ Promise.then 的返回值同样是一个新的 Promise  https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#%E8%BF%94%E5%9B%9E%E5%80%BC  也可以看这个实现细节 https://juejin.cn/post/6850037281206566919#heading-5
